### PR TITLE
C314

### PR DESCRIPTION
**Pull request by cledvina**
Ok, I forgot about the validation stuff for the profile editor.

* The user gets a warning when the resource ID is not unique
* URI resolution checking is performed on resource URI and property URI fields.  If the response is not 'OK'  the error modal appears describing the problem.  Also, there will be a little warning message below the input box that will stay visible.  This happens after moving away from the URI input boxes (on blur as it were.)
* The property types are now stored in verso and populate a select box with the five valid property types.  Currently the selections will need to be edited via the verso api. (perhaps in the following weeks we could add a settings section that deals with configs).  There is a load-type.js script that I pushed to verso.  It will load the types upon booting the server.

### CODE REVIEW

File: source/assets/js/modules/profile/resourcetemplate/controllers/resourceTemplate.controller.js [hunk 1]
```diff
@@ -124,6 +124,40 @@ angular.module('locApp.modules.profile.controllers')
...
+         * Check if URI resolves
+         */
+        $scope.checkURI = function() {
+            $scope.resourceForm.resourceURI.$warn = false;
+            var url = 'server/whichrt?uri=' + $scope.resourceTemplate.resourceURI;
+            $http({
+                method: 'HEAD',
+                url: url
+            })
+            .then(function (response) {
```
Comment (kirkhess) [inline source/assets/js/modules/profile/resourcetemplate/controllers/resourceTemplate.controller.js]: This causes an exception when there is no response - "cannot read property match of undefined"

File: source/assets/js/modules/profile/resourcetemplate/controllers/resourceTemplate.controller.js [hunk 2]
```diff
@@ -124,6 +124,40 @@ angular.module('locApp.modules.profile.controllers')
...
+            }, function (response) {
```
Comment (kirkhess) [inline source/assets/js/modules/profile/resourcetemplate/controllers/resourceTemplate.controller.js]: I think this is an error, so prob. "error" is a better name

Comment (kirkhess) [reply source/assets/js/modules/profile/resourcetemplate/controllers/resourceTemplate.controller.js]: If there's no message there's no alert apparently - see commit 4ddc2d0 below

Comment (kirkhess) [reply source/assets/js/modules/profile/resourcetemplate/controllers/resourceTemplate.controller.js]: 4ddc2d0a7e7f0e99061c8ca46ec474b2aa175dc4

**Comment by cledvina** [comment]
Yeah-- I added those two lines to httpResponse service in order to display the error.  Since the error is not wanted and a pain in the arse, removing those lines makes it better.  We do get an error message in the console, but I can live with that.

**Pull request merged**